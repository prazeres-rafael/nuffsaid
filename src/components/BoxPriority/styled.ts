import styled from "styled-components";
import { palette } from "../../utils/palette";

type Props = {
  $priority: number;
};

export const BoxPriority = styled.div<Props>(({ $priority }) => ({
  padding: 20,
  display: "flex",
  justifyContent: "space-between",
  backgroundColor:
    $priority === 0
      ? palette.messages.error
      : $priority === 1
      ? palette.messages.warning
      : palette.messages.info,
  marginBottom: 10,
}));

export const ButtonClear = styled.a({
  cursor: "pointer",
  left: 50,
});
