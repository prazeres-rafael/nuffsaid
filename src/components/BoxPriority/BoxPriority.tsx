import React, { Fragment, useContext } from "react";
import { ButtonClear, BoxPriority } from "./styled";
import { Message } from "../../Api";
import { MessageContext } from "../../context/MessageContext";

type Props = {
  priority: number;
  messages: Message[];
};

export default function BoxByPriority({
  priority,
  messages,
}: Props): React.ReactElement {
  const { setMessages } = useContext(MessageContext);

  const priorityFilter = (priority: number) => {
    return messages.filter((msgType) => msgType.priority === priority);
  };

  const handleClearItem = (message: string) => {
    const index = messages.findIndex((item) => item.message === message);
    const filteredMessages = [
      ...messages.slice(0, index),
      ...messages.slice(index + 1),
    ];
    setMessages(filteredMessages);
  };

  return (
    <Fragment>
      <p>Count: {priorityFilter(priority).length}</p>
      {priorityFilter(priority).map((msg) => (
        <>
          <BoxPriority $priority={msg.priority} data-testid="box-priority">
            {msg.message}
            <ButtonClear
              onClick={() => handleClearItem(msg.message)}
              data-testid="clear-button"
            >
              Clear
            </ButtonClear>
          </BoxPriority>
        </>
      ))}
    </Fragment>
  );
}
