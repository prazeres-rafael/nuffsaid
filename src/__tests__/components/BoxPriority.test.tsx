import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import BoxPriority from "../../components/BoxPriority/BoxPriority";

const messages = [
  { message: "Et aut quo laboriosam at ut.", priority: 0 },
  { message: "Dolor quo est expedita nostrum corrupti.", priority: 1 },
  { message: "Recusandae iure et.", priority: 2 },
];

describe("BoxPriority", () => {
  it("should match the snapshot", () => {
    const { container } = render(
      <BoxPriority priority={0} messages={messages} />
    );
    expect(container).toMatchSnapshot();
  });

  it("render the correct color when priority is changed", () => {
    const { container } = render(
      <BoxPriority priority={0} messages={messages} />
    );

    expect(screen.getByTestId("box-priority")).toHaveStyle(
      "background-color: #F56236"
    );
  });
});
