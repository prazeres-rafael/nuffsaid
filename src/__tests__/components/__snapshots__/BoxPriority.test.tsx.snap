// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxPriority should match the snapshot 1`] = `
<div>
  <p>
    Count: 
    1
  </p>
  <div
    class="sc-bdvvtL gYHhVl"
    data-testid="box-priority"
  >
    Et aut quo laboriosam at ut.
    <a
      class="sc-gsDKAQ hPlvSa"
      data-testid="clear-button"
    >
      Clear
    </a>
  </div>
</div>
`;
